src/current.rs

```diff
@@ -34,8 +34,11 @@ pub fn clock_for(task_id: TaskId) -> VectorClock {
 }
 
 /// Sets the `tag` field of the current task.
-pub fn set_tag_for_current_task(tag: Tag) {
+/// Returns the `tag` which was there previously.
+pub fn set_tag_for_current_task(tag: Tag) -> Tag {
+    let old_tag = get_tag_for_current_task();
     ExecutionState::set_tag_for_current_task(tag);
```

Comment:
Would be nicer to just let `ExecutionState::set_tag_for_current_task` return the old tag, instead of introducing a local variable and calling `get_tag...`

Reply:
Lol, true, my bad